### PR TITLE
Feature/message display

### LINKED ISSUE #11: Show Group Header in Messages
- [ ] The current day should be displayed as Today.
- [ ] The previous day should be displayed as Yesterday.

### PR DESCRIPTION
**Pull request by test0n3**
* Show reception time for each message
* Group messages by date and show then accordingly
* Provide necessary functions to support dates
* Provide necessary styles for divisions and old messages.

close #11, #10 and #9

### CODE REVIEW

File: public/js/main.js [hunk 1]
```diff
@@ -106,25 +109,78 @@ function initSocket() {
   });
 
   socket.addEventListener('message', event => {
-    const newMessage = JSON.parse(event.data);
```
Comment (DiegoCuevas) [inline public/js/main.js]: why you eliminate the ````const newMessage?

File: public/js/main.js [hunk 2]
```diff
@@ -134,18 +190,35 @@ function renderMessages(messages) {
     }, '');
 }
 
+function prepareMessages(messages) {
+  let text = groupBy(messages, 'id');
+  var html = '';
```
Comment (condef5) [inline public/js/main.js]: Hey bro, you should use `let`.

File: public/js/main.js [hunk 3]
```diff
@@ -134,18 +190,35 @@ function renderMessages(messages) {
...
+    html += renderMessages(filterMessages, curr.length != index + 1);
+  });
+  $listMessage.innerHTML = html;
+}
+
 function renderTime({ id }) {
-  const hours = ('0' + new Date(id).getHours()).slice(-2);
-  const minutes = ('0' + new Date(id).getMinutes()).slice(-2);
-  const seconds = ('0' + new Date(id).getSeconds()).slice(-2);
+  const time = new Date(id);
```
Comment (condef5) [inline public/js/main.js]: Good refactoring

File: public/js/main.js (hunk 1 above)
Comment (DiegoCuevas) [inline public/js/main.js]: why you eliminate ```const newMessage````?

File: public/js/main.js [hunk 4]
```diff
@@ -134,18 +190,35 @@ function renderMessages(messages) {
     }, '');
 }
 
+function prepareMessages(messages) {
```
Comment (condef5) [inline public/js/main.js]: I think that this function should be called renderMessage.

File: public/js/main.js [hunk 5]
```diff
@@ -106,25 +109,78 @@ function initSocket() {
...
+  }, {});
+}
+
+function addDivision(data) {
+  return `<li class="division"><hr /><span>${formatDate(
+    data
+  )}</span><hr /></li>`;
+}
+
 function renderMessages(messages) {
```
Comment (condef5) [inline public/js/main.js]: I think that this function should be called `prepareMessage`

File: public/js/main.js (hunk 1 above)
Comment (condef5) [reply public/js/main.js]: This is my intellectual property.

File: public/js/main.js (hunk 2 above)
Comment (test0n3) [reply public/js/main.js]: Yes my lich.

File: public/js/main.js (hunk 4 above)
Comment (test0n3) [reply public/js/main.js]: Consider it done.

File: public/js/main.js (hunk 5 above)
Comment (test0n3) [reply public/js/main.js]: Consider it done.

**Comment by test0n3** [comment]
This PR closed the following close #11, close #10 , close #9

**Pull request merged**